tests/containers/multi_runtime.pm

```diff
@@ -0,0 +1,118 @@
+# SUSE's openQA tests
+#
+# Copyright 2024-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Validate docker and podman network interoperability
+# - validate if podman and docker networking works together
+# - check if firewalld doesn't break either
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base qw(containers::basetest);
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use utils;
+use containers::common qw(install_packages);
+use Utils::Logging 'save_and_upload_log';
+
+my @containers;
+my $test_image = "registry.opensuse.org/opensuse/nginx";
+
+sub run_tests {
+    my $test_url = "http://www.opensuse.org";
+    my $volumes = '-v $HOME/nginx/nginx.conf:/etc/nginx/nginx.conf:ro,z -v $HOME/nginx:/usr/share/nginx/html:ro,z';
+    my $curl_opts = "-svL -o /dev/null";
+    my $port = 8000;
+
+    # Create rootfull & rootless containers on both podman & docker
+    for my $runtime ("podman", "docker") {
+        for my $type ("root", "rootless") {
+            my $name = "test_${type}_${runtime}";
+            my $sudo = ($type eq "root") ? "sudo" : "";
+            script_retry("$sudo $runtime pull $test_image", retry => 3, delay => 60, timeout => 180);
+            assert_script_run "$sudo $runtime run -d --name $name -p $port:80 $volumes $test_image";
+            push @containers, {name => $name, runtime => $runtime, sudo => $sudo, port => $port};
+            $port++;
+        }
+    }
+
+    # Test connectivity to all containers on localhost
+    for my $container (@containers) {
+        assert_script_run "curl -4 $curl_opts http://127.0.0.1:$container->{port}",
+          fail_message => "failed IPv4 localhost test for $container->{name}";
+    }
+
+    # Test container connectivity to the outside world
+    for my $container (@containers) {
+        assert_script_run "$container->{sudo} $container->{runtime} exec $container->{name} curl -4 $curl_opts $test_url",
+          fail_message => "failed IPv4 Internet test for $container->{name}";
+    }
+}
+
+sub run {
+    select_serial_terminal;
+
+    my @pkgs = ("docker", "podman");
+    install_packages(@pkgs);
+
+    systemctl "enable --now docker";
+
+    record_info("docker root", script_output("docker info"));
+    record_info("podman root", script_output("podman info"));
+
+    # Needed to avoid:
+    # WARNING: COMMAND_FAILED: '/sbin/iptables -t nat -F DOCKER' failed: iptables: No chain/target/match by that name.
+    systemctl "restart firewalld";
+    systemctl "status firewalld";
+
+    assert_script_run "echo '$testapi::username ALL=(ALL:ALL) NOPASSWD: ALL' | tee -a /etc/sudoers.d/nopasswd";
```

Comment:
If we do such invasive configuration on the system, either try to avoid it or set the snapshot restore after test module execution, so that other test modules will not be affected by it.

Reply:
I plan this module to be run alone. But we need password-less sudo to make the implementation easier without having to switch between root & user terminals.

Reply:
Yes that is true. But we can add `always_rollback => 1` and make openQA clean up after the test module.

Reply:
Where shall I add this parameter?

Reply:
See https://open.qa/docs/#_test_flags:

```
sub test_flags () {
    return {always_rollback => 1};
}

Reply:
Fixed